pandas/tests/window/test_rolling.py

```diff
@@ -81,8 +90,39 @@ def test_invalid_constructor(frame_or_series, w):
         c(window=2, min_periods=1, center=w)
 
 
+@pytest.mark.parametrize(
+    "window",
+    [
+        timedelta(days=3),
+        Timedelta(days=3),
+        "3D",
+        ExpandingIndexer(window_size=3),
+        ExponentialMovingWindowIndexer(window_size=3),
+        GroupbyIndexer(window_size=3),
+        VariableOffsetWindowIndexer(
+            index=date_range("2015-12-26", periods=3), offset=BusinessDay(1)
+        ),
+        VariableWindowIndexer(window_size=3),
+    ],
+)
+def test_constructor_step_not_implemented(window, step):
+    # GH 15354
+    n = 10
+    df = DataFrame(
+        {"value": np.arange(n)},
+        index=date_range("2015-12-24", periods=n, freq="D"),
+    )
+    f = lambda: df.rolling(window=window, step=step)
```

Comment:
Thanks this is better!

Could you test the actual constructors raising though for expanding/groupby rolling/ewm the indexers are not exposed to the user? 

i.e. `df.expanding(step=3)`, `df.ewm(step=3)`, `df.groupby(key).rolling(step3)` should raise.

Also are there tests for `df.rolling.cov/corr` raising?

Reply:
I've uploaded tests that cover these cases.